packages/libraries/yoga/src/index.ts

```diff
@@ -147,46 +135,66 @@ export function useHive(clientOrOptions: HiveClient | HivePluginOptions): Plugin
         },
       };
     },
-    onResultProcess(context) {
-      const record = contextualCache.get(context.serverContext);
+    onResultProcess({ serverContext, result }) {
+      const record = contextualCache.get(serverContext);
 
-      if (
-        !record ||
-        Array.isArray(context.result) ||
-        isAsyncIterable(context.result) ||
-        record.executionArgs
-      ) {
+      if (!record || Array.isArray(result) || isAsyncIterable(result) || record.executionArgs) {
         return;
       }
 
       // Report if execution was skipped due to response cache ( Symbol.for('servedFromResponseCache') in context.result)
       if (
         record.paramsArgs.query &&
         latestSchema &&
-        Symbol.for('servedFromResponseCache') in context.result
+        Symbol.for('servedFromResponseCache') in result
       ) {
         try {
           let document = parsedDocumentCache.get(record.paramsArgs.query);
           if (document === undefined) {
             document = parse(record.paramsArgs.query);
             parsedDocumentCache.set(record.paramsArgs.query, document);
           }
-          void record.callback(
-            {
-              document,
-              schema: latestSchema,
-              variableValues: record.paramsArgs.variables,
-              operationName: record.paramsArgs.operationName,
-            },
-            context.result,
-            record.experimental__documentId,
+          serverContext.waitUntil(
+            record.callback(
+              {
+                document,
+                schema: latestSchema,
+                variableValues: record.paramsArgs.variables,
+                operationName: record.paramsArgs.operationName,
+              },
+              result,
+              record.experimental__documentId,
+            ),
           );
         } catch (err) {
-          console.error(err);
+          yoga.logger.error(err);
         }
       }
     },
     onPluginInit({ addPlugin }) {
+      hive = isHiveClient(clientOrOptions)
+        ? clientOrOptions
+        : createHive({
+            ...clientOrOptions,
+            agent: clientOrOptions.agent
+              ? {
+                  logger: {
+                    // Hive Plugin should respect the given Yoga logger
+                    error: (...args) => yoga.logger.error(...args),
+                    info: (...args) => yoga.logger.info(...args),
+                  },
+                  ...clientOrOptions.agent,
+                  __testing: {
```

Comment:
Maybe lets drop the `__testing` thing and add a `fetch` option on the top level instead?

Reply:
@kamilkisiela What do you think?

Reply:
Yeah. The `__testing` is just used for testing, it's not meant to be used this way. We can do a top level setting.

Reply:
I changed it to be a top level setting. Let me know what do you think?